pkg/plugin/driver.go

```diff
@@ -215,21 +210,44 @@ func (h *Clickhouse) Connect(ctx context.Context, config backend.DataSourceInsta
 		opts.DialContext = dialCtx
 	}
 
+	ctx, cancel := context.WithTimeout(ctx, time.Duration(t)*time.Second)
+	defer cancel()
+
 	db := clickhouse.OpenDB(opts)
 
+	// Set connection pool settings
+	if i, err := strconv.Atoi(settings.ConnMaxLifetime); err == nil {
+		db.SetConnMaxLifetime(time.Duration(i) * time.Minute)
```

Comment:
Nit: I'd suggest having this parsing logic in settings.go instead just so once we're in the driver then the settings object is ready to go without needing to carry out any more operations against it. Feel free to disregard!

Reply:
Let's fix this up in a separate PR... `DialTimeout` and `QueryTimeout` are done the same way and we should fix this up in all cases, and I really want to get this PR out, so let's come back to this later.